### PR TITLE
feat(watcher): support multiply watch options

### PR DESCRIPTION
**Pull request by underfin**
### Description

The `Bundler::watch` api is not suitable for watcher if it is multiply watch options, so here add a new `BindingWatcher`. The binding the multiply `Bundler` to create `BindingWatcher` is impossible, so here pass binding multiply options to create it.

The rollup is create `chokidar` for each `Bundler`, it has more overhead at rolldown, because we need to create thread to listen event. So the `notify` option is used for all `Bundler`, i will check it at next.

### CODE REVIEW

**Comment by underfin** [comment]
* **#3037** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3037?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3037?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#3032** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3032?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/rolldown/src/binding.d.ts [hunk 1]
```diff
@@ -167,18 +167,6 @@ export interface ArrowFunctionsOptions {
   spec?: boolean
 }
 
-export interface ArrowFunctionsOptions {
```
Comment (underfin) [inline packages/rolldown/src/binding.d.ts:170]: It's is duplicated.

File: crates/rolldown_binding/src/bundler.rs [hunk 2]
```diff
@@ -19,6 +18,13 @@ use napi_derive::napi;
 use rolldown::Bundler as NativeBundler;
 use rolldown_error::{BuildDiagnostic, BuildResult, DiagnosticOptions};
 
+#[napi(object, object_to_js = false)]
+pub struct BindingBundlerOption {
```
Comment (hyf0) [inline crates/rolldown_binding/src/bundler.rs]:
```suggestion
pub struct BindingBundlerOptions {
```

Plural would be better.

File: packages/rolldown/src/utils/create-bundler-option.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      outputOptions,
+      onLog,
+      logLevel,
+    )
+
+    // Convert `OutputOptions` to `BindingInputOptions`
+    const bindingOutputOptions = bindingifyOutputOptions(outputOptions)
+
+    return {
+      bundlerOption: {
```
Comment (hyf0) [inline packages/rolldown/src/utils/create-bundler-option.ts]:
```suggestion
      bundlerOptions: {
```

**Review by hyf0 — APPROVED** [review]

**Pull request merged**